### PR TITLE
Resurrect search

### PR DESCRIPTION
**Pull request by iojancode**
This PR resurrect Lucene Search. All code was ready, it was just remove comments and test.

I had to use ibmcom/couchdb3 docker image for two main reasons:
1) CouchDB official images are not ready yet with clouseau, but this feature is almost ready apache/couchdb-docker#8
2) Lucene Search is available since couchdb3 https://docs.couchdb.org/en/stable/ddocs/search.html

### CODE REVIEW

File: azure-pipelines-cd.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ pr: none
 resources:
   containers:
   - container: couchdb
-    image: couchdb:2.3.1
+    image: ibmcom/couchdb3:3.1.0
```
Comment (danielwertheim) [inline azure-pipelines-cd.yml:19]: The public one now supports 3.1

File: azure-pipelines-ci.yml [hunk 2]
```diff
@@ -20,7 +20,7 @@ pr:
 resources:
   containers:
   - container: couchdb
-    image: couchdb:2.3.1
+    image: ibmcom/couchdb3:3.1.0
```
Comment (danielwertheim) [inline azure-pipelines-ci.yml:23]: The public one now supports 3.1

File: docker-compose.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 version: '3'
 services:
   couchdb:
-    image: "couchdb:2.3.1"
+    image: "ibmcom/couchdb3:3.1.0"
```
Comment (danielwertheim) [inline docker-compose.yml:4]: The public one now supports 3.1

**Comment by iojancode** [comment]
Hi, thanks for taking the time to review this.
Official image supporting search is going to be named `couchdb:3.1.0-ubi-clouseau`, ready at https://github.com/apache/couchdb-docker/blob/master/3.1.0-ubi-clouseau/Dockerfile . But it has not been built and published to docker registry yet, so that's why I had to use ibmcom image.

**Pull request merged**